pkg/services/object/acl/acl.go

```diff
@@ -194,14 +194,19 @@ func (c *Checker) CheckEACL(msg any, reqInfo v2.RequestInfo) error {
 		eaclRole = eaclSDK.RoleOthers
 	}
 
-	action, _ := c.validator.CalculateAction(new(eaclSDK.ValidationUnit).
+	vu := new(eaclSDK.ValidationUnit).
 		WithRole(eaclRole).
 		WithOperation(eaclSDK.Operation(reqInfo.Operation())).
 		WithContainerID(&cnr).
 		WithSenderKey(reqInfo.SenderKey()).
 		WithHeaderSource(hdrSrc).
-		WithEACLTable(&table),
-	)
+		WithEACLTable(&table)
+
+	if sa := reqInfo.SenderAccount(); sa != nil && !sa.IsZero() {
```

Comment:
can it ever be nil? seems like we can get rid of a pointer

Reply:
missing sender is a strange thing to me, an error should appear earlier, no?

Reply:
Before this change, the account wasn't represented at all. It was only `key []byte`. According to the code, at this point, it can't be nil, but in the future - nobody knows.

Do you suggest removing the pointer?